sql/core/src/test/scala/org/apache/spark/sql/ConfigBehaviorSuite.scala

```diff
@@ -58,7 +58,7 @@ class ConfigBehaviorSuite extends QueryTest with SharedSQLContext {
       withSQLConf(SQLConf.RANGE_EXCHANGE_SAMPLE_SIZE_PER_PARTITION.key -> "1") {
         // If we only sample one point, the range boundaries will be pretty bad and the
         // chi-sq value would be very high.
-        assert(computeChiSquareTest() > 1000)
+        assert(computeChiSquareTest() > 300)
```

Comment:
@rxin, the difference that is causing this to fail is that `rdd.id` is 15 instead of 14 because of the change to `getByteArrayRdd`. That id is used to seed the random generator and the chi-sq value isn't high enough. The previous value must have been unusually high. With so few data points, this can vary quite a bit so I think changing the bound to 300 is a good fix.

Reply:
Updating the SparkPlan code to avoid creating unnecessary RDDs fixed this test because the same random seed is used. But, I think we should keep this change so we don't have to track down this problem again in the future. This bound is perfectly safe because the balanced chi-sq value is 10, while the unbalanced is much larger.